examples/xoxo/xoxo.py

```diff
@@ -0,0 +1,183 @@
+from pysmt.shortcuts import FreshSymbol, Symbol, String, LE, GE, Int, Not, Or, And, Equals, Plus, Solver, StrContains
+from pysmt.typing import INT, STRING
+from enum import Enum
+import logging
+logger = logging.getLogger('xoxo')
+logger.setLevel(logging.DEBUG)
+ch = logging.StreamHandler()
+ch.setLevel(logging.INFO) # change to DEBUG for info on assertions
+logger.addHandler(ch)
+
+class Cell(Enum):
+    x = 1
+    o = 10
+    s = 0
+
+DEBUG = False
+turns = 0
+sq_size = 3
+test = 'tests/test1.txt'
+test = 'tests/blank.txt'
+
+# create symbols for the board
+board = [[FreshSymbol(INT) for _ in xrange(sq_size)]
```

Comment:
Any reason why you are not using BitVectors?

Reply:
My reasoning was to have a board that contained both players moves (X is 1, O is 10), and had an easy method to:

1. know if a win happened. So if the line adds to 3 or 30.
2. have a way to constrain the solver to look for a move on the next turn. So I can say the total value of the board should be 33, that means 3 turns have been played each.

How could I do this with bitvectors?

Reply:
Arithmetics in BV works the same way, so that you can do the same. Alternatively, you could optimize the representation to be more compact. This needs some more thoughts, but for example, if you have a 4 bits word, you can use to left-most for O and the right-most for X.
11XX is a win for O, XX11 is a win for X. A full game is the sum that yields 1111.
The idea is the same, but now you are thinking in base 10, and you can do the same in base 3.

To check whether base 3 is enough for the other constraints that you have.

Reply:
using 4 bit length vectors now